Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -158,4 +158,66 @@ initialize addLinter missingEndLinter
 
 end MissingEnd
 
+/-!
+# The `cdot` linter
+
+The `cdot` linter is a syntax-linter that flags uses of the "cdot" `·` that are achieved
+by typing a character different form `·`.
+For instance, a "plain" dot `.` is allowed syntax, but is flagged by the linter.
+-/
+
+/--
+The `cdot` linter flags uses of the "cdot" `·` that are achieved by typing a character
+different form `·`.
+For instance, a "plain" dot `.` is allowed syntax, but is flagged by the linter. -/
+register_option linter.cdot : Bool := {
+  defValue := true
+  descr := "enable the `cdot` linter"
+}
+
+/-- `isCDot? stx` checks whether `stx` is a `Syntax` node corresponding to a `cdot` typed with
+the character `·`. -/
+def isCDot? : Syntax → Bool
+  | .node _ ``cdotTk #[.node _ `patternIgnore #[.node _ _ #[.atom _ v]]] => v == "·"
+  | .node _ ``Lean.Parser.Term.cdot #[.atom _ v] => v == "·"
+  | _ => false
+
+/--
+`findDot stx` extracts from `stx` the syntax nodes of `kind` `Lean.Parser.Term.cdot` or `cdotTk`. -/
+partial
+def findDot : Syntax → Array Syntax
```

Comment:
Can you call this something with "cdot", please?
(I'm unsure how to capitalise names with cdot in it. I may give that bikeshed more thought after lunch.)

Reply:
Renamed to `findCDot`.  My instinct would be to use the same capitalization for `C` and `D`, like `ZMod`, `EReal`, `PUnit`,...

Reply:
Works for me!